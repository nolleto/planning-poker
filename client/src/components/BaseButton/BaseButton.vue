<template>
  <button
    :class="[
      bootstrap.btn,
      {
        [bootstrap.btnPrimary]: variant === 'primary',
        [bootstrap.btnSecondary]: variant === 'secondary',
        [bootstrap.btnSuccess]: variant === 'success',
        [bootstrap.btnDanger]: variant === 'danger',
        [bootstrap.btnWarning]: variant === 'warning',
        [bootstrap.btnInfo]: variant === 'info',
        [bootstrap.btnLight]: variant === 'light',
        [bootstrap.btnDark]: variant === 'dark',
        [bootstrap.btnLink]: variant === 'link'
      }
    ]"
    :disabled="disabled"
  >
    <slot></slot>
  </button>
</template>

<script>
import styles from '@/styles'

export default {
  name: 'BaseButton',

  props: {
    variant: {
      type: String,
      validator (value) {
        return [
          'primary',
          'secondary',
          'success',
          'danger',
          'warning',
          'info',
          'light',
          'dark',
          'link'
        ].indexOf(value) !== -1
      }
    },

    disabled: {
      type: Boolean
    }
  },

  computed: {
    bootstrap: () => styles.bootstrap
  }
}
</script>
