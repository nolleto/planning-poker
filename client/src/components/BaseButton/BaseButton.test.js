import BaseButton from '.'
import { shallowMount } from '@vue/test-utils'

const shallowComponent = () => {
  return shallowMount(BaseButton)
}

describe('BaseButton component', () => {
  let wrapper

  describe('Snapshots', () => {
    beforeEach(() => {
      wrapper = shallowComponent()
    })

    it('matches', () => {
      expect(wrapper.element).toMatchSnapshot()
    })

    describe('When is disabled', () => {
      beforeEach(() => {
        wrapper.setProps({ disabled: true })
      })

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })

    describe('When "variant" is "primary"', () => {
      beforeEach(() => {
        wrapper.setProps({ variant: 'primary' })
      })

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })

    describe('When "variant" is "secondary"', () => {
      beforeEach(() => {
        wrapper.setProps({ variant: 'secondary' })
      })

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })

    describe('When "variant" is "success"', () => {
      beforeEach(() => {
        wrapper.setProps({ variant: 'success' })
      })

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })

    describe('When "variant" is "danger"', () => {
      beforeEach(() => {
        wrapper.setProps({ variant: 'danger' })
      })

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })

    describe('When "variant" is "warning"', () => {
      beforeEach(() => {
        wrapper.setProps({ variant: 'warning' })
      })

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })

    describe('When "variant" is "info"', () => {
      beforeEach(() => {
        wrapper.setProps({ variant: 'info' })
      })

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })

    describe('When "variant" is "light"', () => {
      beforeEach(() => {
        wrapper.setProps({ variant: 'light' })
      })

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })

    describe('When "variant" is "dark"', () => {
      beforeEach(() => {
        wrapper.setProps({ variant: 'dark' })
      })

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })

    describe('When "variant" is "link"', () => {
      beforeEach(() => {
        wrapper.setProps({ variant: 'link' })
      })

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })
})
