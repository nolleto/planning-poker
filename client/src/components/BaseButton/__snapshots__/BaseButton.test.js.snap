// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton component Snapshots When "variant" is "danger" matches 1`] = `
<button
  class="btn btnDanger"
/>
`;

exports[`BaseButton component Snapshots When "variant" is "dark" matches 1`] = `
<button
  class="btn btnDark"
/>
`;

exports[`BaseButton component Snapshots When "variant" is "info" matches 1`] = `
<button
  class="btn btnInfo"
/>
`;

exports[`BaseButton component Snapshots When "variant" is "light" matches 1`] = `
<button
  class="btn btnLight"
/>
`;

exports[`BaseButton component Snapshots When "variant" is "link" matches 1`] = `
<button
  class="btn btnLink"
/>
`;

exports[`BaseButton component Snapshots When "variant" is "primary" matches 1`] = `
<button
  class="btn btnPrimary"
/>
`;

exports[`BaseButton component Snapshots When "variant" is "secondary" matches 1`] = `
<button
  class="btn btnSecondary"
/>
`;

exports[`BaseButton component Snapshots When "variant" is "success" matches 1`] = `
<button
  class="btn btnSuccess"
/>
`;

exports[`BaseButton component Snapshots When "variant" is "warning" matches 1`] = `
<button
  class="btn btnWarning"
/>
`;

exports[`BaseButton component Snapshots When is disabled matches 1`] = `
<button
  class="btn"
  disabled="disabled"
/>
`;

exports[`BaseButton component Snapshots matches 1`] = `
<button
  class="btn"
/>
`;
